pkg/registry/core/pod/strategy.go

```diff
@@ -89,9 +93,31 @@ func (podStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func isUpdatingUninitializedPod(old runtime.Object) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
+		return false, nil
+	}
+	oldMeta, err := meta.Accessor(old)
+	if err != nil {
+		return false, err
+	}
+	oldInitializers := oldMeta.GetInitializers()
+	if oldInitializers != nil && len(oldInitializers.Pending) != 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
 // ValidateUpdate is the default update validation for an end user.
 func (podStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
 	errorList := validation.ValidatePod(obj.(*api.Pod))
+	uninitializedUpdate, err := isUpdatingUninitializedPod(old)
```

Comment:
@liggitt I just saw the code you posted. Do you have a strong preference using the `switch` statement?

Reply:
I find it easier to read, but don't feel strongly